azurerm/resource_arm_container_group.go

```diff
@@ -146,16 +147,48 @@ func resourceArmContainerGroup() *schema.Resource {
 						},
 
 						"port": {
-							Type:         schema.TypeInt,
-							Optional:     true,
-							ForceNew:     true,
-							ValidateFunc: validation.IntBetween(1, 65535),
+							Type:          schema.TypeInt,
+							Optional:      true,
+							ForceNew:      true,
+							ConflictsWith: []string{"container.0.ports"},
+							Deprecated:    "Use `ports` instead.",
+							ValidateFunc:  validate.PortNumber,
+						},
+
+						"ports": {
+							Type:          schema.TypeList,
```

Comment:
We should probably use a `TypeSet` here as order doesn't matter? at we shouldn't allow duplicates.